src/main/java/org/elasticsearch/search/aggregations/reducers/derivative/DerivativeBuilder.java

```diff
@@ -45,6 +46,16 @@ public DerivativeBuilder gapPolicy(GapPolicy gapPolicy) {
         return this;
     }
 
+    public DerivativeBuilder units(String units) {
+        this.units = units;
+        return this;
+    }
+
+    public DerivativeBuilder units(DateHistogramInterval units) {
+        this.units = units.toString();
+        return this;
+    }
```

Comment:
I see. Maybe we should consider adding a wrapper similar to `TimeValue` to avoid string manipulation. But this is no new issue, let's defer...